packages/hooks/src/useTransferMessageBetweenWalletAndTradingApp.ts

```diff
@@ -0,0 +1,75 @@
+import useExchangeRate from './useExchangeRate';
+import useWalletTransfer from './useWalletTransfer';
+
+type TTransferMessageBetweenWalletAndTradingApp =
+    | {
+          code: 'WalletToTradingAppDailyLimit';
+          is_first_transfer: boolean;
+          limit: number;
+          currency: string;
+          type: 'error' | 'info' | 'success';
+      }
+    | {
+          code: 'DemoWalletToTradingAppDailyLimit';
+          is_first_transfer: boolean;
+          limit: number;
+          currency: string;
+          type: 'error' | 'info' | 'success';
+      };
+
+const tradingAccountMapper = (account_type: string, currency_type: string): string => {
+    if (account_type === 'trading') {
+        if (currency_type === 'demo') return 'virtual';
+        return 'dtrade';
+    }
+    return account_type;
+};
+
+/**
+ * Generates and returns the list of all messages to be shown for transfer between wallets and trading accounts linked to it.
+ * @param from_account - Information of the source account received from the useWalletsList hook.
+ * @param to_account - Information of the destination account received from the useWalletsList hook.
+ */
+const useTransferMessageBetweenWalletAndTradingApp = (
+    from_account: Partial<ReturnType<typeof useWalletTransfer>['from_account']>,
+    to_account: Partial<ReturnType<typeof useWalletTransfer>['to_account']>,
+    account_limits: any // TODO: add type of account_limits when the new schema is ready.
```

Comment:
Still new to this so I'm just asking some questions for clarification. How is the error handling carried out or is error handling needed for this process, let's say for example if the parameters such as from_account or to_account are not received as expected from the useWalletsList hook

Reply:
Hi @lubega-deriv,
The params for this hook are coming from the `useAuthorize` hook which in-turn is making `authorize` API call. Then the response of this call is passed to `useWalletList` which transforms the data into `from_account` and `to_account`. So if anything might go wrong from the backend, the app will show the `server hit bump` error screen.
If I am right, the error handling of the response is backend's concern.

Please correct me if I am wrong @heorhi-deriv.

Reply:
alright understood, thanks for the clarification!